src/components/activity-page/managed-interactive/managed-interactive.tsx

```diff
@@ -1,40 +1,59 @@
-import React, { useState, useEffect, useRef } from "react";
+import React from "react";
 import { IframeRuntime } from "./iframe-runtime";
+import useResizeObserver from "@react-hook/resize-observer";
 
 interface IProps {
   embeddable: any;
   questionNumber?: number;
 }
 
+const kDefaultAspectRatio = 4 / 3;
+
 export const ManagedInteractive: React.FC<IProps> = (props) => {
 
     const handleNewInteractiveState = (state: any) => {
       // TODO: handle interactive state
     };
 
-    const [initialHeight, setInitialHeight] = useState(0);
-    const divRef: any = useRef();
-    useEffect(() => {
-      if (divRef) {
-        const height = divRef.current.getBoundingClientRect().width * 3 / 4;
-        setInitialHeight(height);
-      }
-    }, [divRef]);
-
     const { embeddable, questionNumber } = props;
     const questionName = embeddable.name ? `: ${embeddable.name}` : "";
-    const url = embeddable.url
-                ? embeddable.url
-                : (embeddable.library_interactive.data.base_url ? embeddable.library_interactive.data.base_url : "");
+    // in older iframe interactive embeddables, we get url, native_width, native_height, etc. directly off
+    // of the embeddable object. On newer managed/library interactives, this data is in library_interactive.data.
+    const embeddableData = embeddable.library_interactive?.data ? embeddable.library_interactive.data : embeddable;
+    const url = embeddableData.base_url ? embeddableData.base_url : embeddableData.url ? embeddableData.url : "";
+    // TODO: handle different aspect ration methods
+    // const aspectRatioMethod = data.aspect_ratio_method ? data.aspect_ratio_method : "";
+    const nativeHeight = embeddableData.native_height ? embeddableData.native_height : 0;
+    const nativeWidth = embeddableData.native_width ? embeddableData.native_width : 0;
+    const aspectRatio = nativeHeight && nativeWidth ? nativeWidth / nativeHeight : kDefaultAspectRatio;
```

Comment:
Regarding the TODO above:
- if the method is 'DEFAULT' then the nativeHeight and nativeWidth are ignored and the `4/3` is used. Unless the interactive provides its own aspectRatio in supported features or its own height with the height message.
- if the method is 'MANUAL' then the nativeHeight and nativeWidth are required, and the supported features aspectRatio is ignored. The height message is still allowed though and that will override the nativeHeight and nativeWidth aspect ratio.
- if the method is 'MAX' then the nativeHeight and nativeWidth are ignored. The supported features aspectRatio is ignored. The height message is still allowed.  If the height message is not sent, then the width is '100%' and the height is 100 percent of the available visible area. If there is no header or footer needed then in css this would be `100vh`.